### ISSUE TITLE
[Question] How to auto-enable Bitcode for Pods project?

### ISSUE THREAD

**Issue by derwaldgeist** [googlesamples/unity-jar-resolver#408]
* Unity editor version: 2019.4.12f1
  * External Dependency Manager version: 1.2.135.0
  * Source you installed EDM4U: Facebook SDK (.unitypackage)
  * Features in External Dependency Manager in use: I guess all
  * Plugins SDK in use: Facebook
  * Platform you are using the Unity editor on: Mac

After I installed the Facebook Unity SDK, my Unity project won't archive any more because of this Bitcode-related problem:

> bitcode bundle could not be generated because '/Users/tom/Desktop/Marble Builds/marble.stage.ios.1.17/Libraries/Plugins/iOS/libBackgroundLocationPlugin.a(Pods-BackgroundLocationPlugin-dummy.o)' was built without full bitcode. All object files and libraries for bitcode must be generated from Xcode Archive or Install build file '/Users/tom/Desktop/Marble Builds/marble.stage.ios.1.17/Libraries/Plugins/iOS/libBackgroundLocationPlugin.a' for architecture arm64

I was able to track this down to two issues:

1. Although Bitcode had already been enabled during compilation of the library (by setting the ENABLE_BITCODE flag and the `-fembed-bitcode` flag), this wasn't enough. The library also uses Pods, and I had to enable the `ENABLE_BITCODE` flag to its Pods project as well to make it work. Strange enough, this is *not* necessary if the Facebook SDK is not present. I guess it has something to do with the different two-project setup in the Unity project (main Unity-iPhone + Pods) which was not the case without the SDK?

2. On the Pods of the project generated by Unity, the `ENABLE_BITCODE` flag is set to `false` after the export. Only if I set this flag to `true` as well, archiving works as expected.

My actual question is related to this second part, because I was not able to find a way to automate this. I tried to achieve this with a PostBuild script, but this did not work:

```cs
[PostProcessBuild]
public static void OnPostProcessBuildIos(BuildTarget target, string pathToBuiltProject) {
	string projPath = pathToBuiltProject + "/Pods/Pods.xcodeproj/project.pbxproj";
	PBXProject proj = new PBXProject();
	proj.ReadFromString(File.ReadAllText(projPath));
	string targetGuid = proj.GetUnityFrameworkTargetGuid();
	proj.AddBuildProperty(targetGuid, "ENABLE_BITCODE", "true");
	File.WriteAllText(projPath, proj.WriteToString());
}
```

The problem is that the `Pods.xcodeproj` file is not available at the time my post build script runs, so it won't be able to patch the flag.

Is there any other way to force the resolver to enable the Bitcode flag? I don't want to do this manually after every build. I was hoping I could enforce this by setting the `enableBitcode` flags in the `Dependencies.xml` file of the Facebook SDK to true, but this did not work either.

**Comment by derwaldgeist** [comment]
> There's a [Readme section](https://github.com/googlesamples/unity-jar-resolver/blob/825901fa297065d651709fdc34cc5433410c8869/README.md#ios-resolver-usage) detailing how to enable BITCODE for pods by altering the dependency file. Does this help you at all?

I tried exactly this, by patching the SDK's dependency file (where the bitcode flag wasn't set). But this did not work. And AFAIR, this was despite all of their libraries *were* bitcode-enabled.

**Comment by derwaldgeist** [comment]
BTW: I got a notification by Facebook in the meantime who stated that the issue was fixed in their most recent SDK version. But I haven't had the time to check this back.

**Comment by vimanyu** [comment]
Hi @derwaldgeist,
Which version of Facebook plugin were you using when you ran into the problem?

I tried to reproduce this locally on my end,
- Create a new Unity 2019 project
- Create XCode archive  // Success
- Install Facebook SDK Unity plugin (8.1.1)
- Create XCode archive // Success

Didnt see any bitcode errors.

**Comment by derwaldgeist** [comment]
Good question. It was the latest available on their website at that time. Based on what I can find in my project sources, it seems to be 8.1.0. At least the wrapper is named facebook-android-wrapper-8.1.0.aar.

They said they have fixed the problem in the meantime, so this might have already landed in 8.1.1. I kept this open because I was wondering if there is a way to override the Bitcode settings using the resolver if the developer did something wrong.

**Comment by vimanyu** [comment]
I see. I just downloaded 8.1.0 and couldn't reproduce this problem on my end with that too.
- Create a new Unity 2019 project
- Build XCode project from Unity // Success
- Install Facebook SDK Unity plugin (8.1.0)
- Build XCode project from Unity  // Success

Ideally, fixing all *Dependencies.xml with bitcode enabled should have done the trick. It will be nice to reproduce this locally before answering your question.

When you have some time, could you please provide a minimal set of repro steps to create that error?
